sudoku_gpso.py

```diff
@@ -98,10 +111,29 @@ def crossover2(self, global_best, weights) -> None:
         i dopiero wtedy zwracana jest cała plansza.
          
         """
-        # TODO: implementacja drugiej wersji krzyżowania
-        next_pos = []
-        self.update_curr_pos(next_pos)
         
+        parent_tup = (     # kolejność do ustalenia
+            deepcopy(self.curr_pos),
+            deepcopy(self.local_best_position),
+            deepcopy(global_best).tolist()
+        )
+        
+        for row in range(9):
+            mask = rnd.choices(parent_tup, weights=weights, k=9)    # maska wybierana ponownie dla każdego z wierszy - zweryfikować,
```

Comment:
to bym wrzuciła do metody get_mask - co prawda w artykule troche jest inna reprezentacja, ale takie podejście wydaje się spoko
jeżeli chodzi o to czy za każdym razem tworzymy maskę - na razie bym zostawiła tak jak jest, ale możemy przetestować obydwie wersje i zobaczyć która lepiej działa

Reply:
pyatnie jeszcze czy parametr weights zadziała w przypadku liczb z zakresu <0,1> ? nie trzeba ich przemnożyć do liczb całkowitych?